backend/app/api/api_v1/routers/users.py

```diff
@@ -1,85 +1,52 @@
+    deactivate_user,
+)
+from app.db.schemas.user import User, UserBase
 from app.db.session import get_db
-from fastapi import APIRouter, Depends, Request, Response
 
 users_router = r = APIRouter()
 
-
-@r.get(
-    "/users",
-    response_model=t.List[User],
-    response_model_exclude_none=True,
-)
-async def users_list(
-    response: Response,
-    db=Depends(get_db),
-    current_user=Depends(get_current_active_superuser),
-):
-    """Gets all users"""
-    users = get_users(db)
-    # This is necessary for react-admin to work
-    response.headers["Content-Range"] = f"0-9/{len(users)}"
-    return users
+logger = logging.getLogger(__file__)
 
 
 @r.get("/users/me", response_model=User, response_model_exclude_none=True)
 async def user_me(current_user=Depends(get_current_active_user)):
     """Gets own user"""
-    return current_user
-
-
-@r.get(
-    "/users/{user_id}",
-    response_model=User,
-    response_model_exclude_none=True,
-)
-async def user_details(
-    request: Request,
-    user_id: int,
-    db=Depends(get_db),
-    current_user=Depends(get_current_active_superuser),
-):
-    """Gets any user details"""
-    user = get_user(db, user_id)
-    return user
-    # return encoders.jsonable_encoder(
-    #     user, skip_defaults=True, exclude_none=True,
-    # )
 
-
-@r.post("/users", response_model=User, response_model_exclude_none=True)
-async def user_create(
-    request: Request,
-    user: UserCreate,
-    db=Depends(get_db),
-    current_user=Depends(get_current_active_superuser),
-):
-    """Creates a new user"""
-    return create_user(db, user)
+    return current_user
 
 
-@r.put("/users/{user_id}", response_model=User, response_model_exclude_none=True)
+@r.put("/users/me", response_model=User, response_model_exclude_none=True)
 async def user_edit(
     request: Request,
-    user_id: int,
-    user: UserEdit,
+    user: UserBase,
     db=Depends(get_db),
-    current_user=Depends(get_current_active_superuser),
+    current_user=Depends(get_current_active_user),
 ):
     """Updates existing user"""
-    return edit_user(db, user_id, user)
 
+    updated_user = edit_user(db, current_user.id, user)
+    send_email(EmailType.account_changed, updated_user)
+    return updated_user
 
-@r.delete("/users/{user_id}", response_model=User, response_model_exclude_none=True)
-async def user_delete(
+
+@r.delete("/passwords/me", response_model=bool, response_model_exclude_none=True)
+async def delete_password(
     request: Request,
-    user_id: int,
     db=Depends(get_db),
-    current_user=Depends(get_current_active_superuser),
+    current_user=Depends(get_current_active_user),
 ):
-    """Deletes existing user"""
-    return delete_user(db, user_id)
+    """Deletes a password, and kicks off password-reset flow."""
+
+    deactivate_user(db, current_user.id)
+    activation_token = create_password_reset_token(db, current_user.id)
+    send_email(EmailType.password_reset_requested, current_user, activation_token)
```

Comment:
There seem to be a couple of issues here:
- It looks like a user must be logged in in order to request that their password is reset because they've forgotten it, which seems like it could be an issue.
- The password reset form should be unauthenticated, take an email & if valid kick off a password reset flow for a user, this shouldn't immediately disable a user or it's open for abuse.

Reply:
True 👍🏻 Will modify.